com.unity.visualeffectgraph/Editor/GraphView/Views/VFXView.cs

```diff
@@ -2127,10 +2127,15 @@ public override void RemoveFromSelection(ISelectable selectable)
 
         public override void ClearSelection()
         {
-            bool selectionEmpty = selection.Count() == 0;
             base.ClearSelection();
-            if (!selectionEmpty)
-                UpdateGlobalSelection();
+            Selection.objects = controller != null && controller.model != null
+                ? new[] { controller.model.isSubgraph ? controller.model.subgraph : (VisualEffectObject)controller.model.asset }
+                : null;
```

Comment:
This behavior is still being debated because the project browser also browse the the VFX asset when selected which can be annoying for the user.
Anyway, it's not a bug and it's consistent the way selection works in Unity.
Although there will probably be another PR to change a bit this behavior.